pkg/util/json.go

```diff
@@ -1,4 +1,112 @@
 package util
 
+import (
+	"encoding/json"
+
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/grafana/grafana/pkg/util/errutil"
+)
+
 // DynMap defines a dynamic map interface.
 type DynMap map[string]any
+
+var (
+	// ErrEmptyJSON is an error for empty attribute in JSON.
+	ErrEmptyJSON = errutil.NewBase(errutil.StatusBadRequest,
+		"json-missing-body", errutil.WithPublicMessage("Empty JSON provided"))
+
+	// ErrNoAttributePathSpecified is an error for no attribute path specified.
+	ErrNoAttributePathSpecified = errutil.NewBase(errutil.StatusBadRequest,
+		"json-no-attribute-path-specified", errutil.WithPublicMessage("No attribute path specified"))
+
+	// ErrFailedToUnmarshalJSON is an error for failure in unmarshalling JSON.
+	ErrFailedToUnmarshalJSON = errutil.NewBase(errutil.StatusBadRequest,
+		"json-failed-to-unmarshal", errutil.WithPublicMessage("Failed to unmarshal JSON"))
+
+	// ErrFailedToSearchJSON is an error for failure in searching JSON.
+	ErrFailedToSearchJSON = errutil.NewBase(errutil.StatusBadRequest,
+		"json-failed-to-search", errutil.WithPublicMessage("Failed to search JSON with provided path"))
+)
+
+// SearchJSONForGroups searches for groups in a JSON object and returns a string slice.
```

Comment:
```suggestion
// SearchJSONForStringSliceAttr searches for slice attribute in a JSON object and returns a string slice.
```

Reply:
+1, need to also go cleanup the interface{} in the other func